### PR TITLE
Update gRIBI `motivation.md` based on evolution.

### PR DESCRIPTION
**Pull request by robshakir**
```
* (M) doc/motivation.md
  - Describe motivation for changes to liveliness handling.
  - Describe motivation for selection between clients (leader election)
```

The intention of the edits to this document are to ensure that there is not inconsistency in the protocol documentation as we move towards writing a specification.

### CODE REVIEW

File: doc/motivation.md [hunk 1]
```diff
@@ -257,43 +257,42 @@ which will be resolved elsewhere in the RIB.
...
+## Liveliness of Injected Entries
+
+There has been much debate about how entries installed by a routing protocol
+should be dealt with when the session, or neighbour, that provides those entries
+"goes away". Tension exists being approaches that prefer removing that routing
+information, and approaches that prefer to 'fail static' based on the last
+available entries. In order to provide for both approaches within gRIBI, two
+modes of operation are proposed.
+ 
+ * a mode of operation where the entries on the server are not tied to session
```
Comment (xw-g) [inline doc/motivation.md:269]: Where should we specify the detailed spec of this mode (e.g. any timeout on the fail-static routes, how to deal with the fail-static route when client reconnect) ? 

Maybe a TODO here as place holder for further discussion/proposal?

File: doc/motivation.md [hunk 2]
```diff
@@ -257,43 +257,42 @@ which will be resolved elsewhere in the RIB.
...
+   state (as specified by the client at connection time).
```
Comment (xw-g) [inline doc/motivation.md:270]:
```suggestion
   state (fail static).
```

File: doc/motivation.md [hunk 3]
```diff
@@ -257,43 +257,42 @@ which will be resolved elsewhere in the RIB.
...
-are responsible for ensuring consistency of announcements.
-
+## Liveliness of Injected Entries
+
+There has been much debate about how entries installed by a routing protocol
+should be dealt with when the session, or neighbour, that provides those entries
+"goes away". Tension exists being approaches that prefer removing that routing
+information, and approaches that prefer to 'fail static' based on the last
+available entries. In order to provide for both approaches within gRIBI, two
+modes of operation are proposed.
```
Comment (xw-g) [inline doc/motivation.md:267]:
```suggestion
modes of operation are proposed (specified by the client at connection time).
```

**Review by xw-g — CHANGES_REQUESTED** [review]

File: doc/motivation.md (hunk 1 above)
Comment (robshakir) [reply doc/motivation.md:269]: ACK - we should do this in the detailed spec I think. Currently, there is no timeout - an external client can call `Flush`, and these routes are essentially reconciled through `Get` plus `Modify` operations to bring the server in-line with the expected state.

I figured that just keeping this document extremely high-level such that it serves as "why gRIBI?" would be a good start, and have an outline of a spec that I was planning to push and see whether we should work on.

Let me know if this sounds reasonable -- or I can add something further here to describe this at a top-level if we think the doc is incomplete without it :-)

Comment (xw-g) [reply doc/motivation.md:269]: +1 make sense.

**Review by xw-g — APPROVED** [review]

**Pull request merged**